streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java

```diff
@@ -374,25 +385,15 @@ private void assertInternalTopicsGotDeleted(final String intermediateUserTopic)
         expectedRemainingTopicsAfterCleanup.add(OUTPUT_TOPIC_2_RERUN);
         expectedRemainingTopicsAfterCleanup.add("__consumer_offsets");
 
-        Set<String> allTopics;
-        ZkUtils zkUtils = null;
+        Set<String> allTopics = new HashSet<>();
         try {
-            zkUtils = ZkUtils.apply(CLUSTER.zKConnectString(),
-                30000,
-                30000,
-                JaasUtils.isZkSecurityEnabled());
-
-            do {
-                Utils.sleep(100);
-                allTopics = new HashSet<>();
-                allTopics.addAll(scala.collection.JavaConversions.seqAsJavaList(zkUtils.getAllTopics()));
-            } while (allTopics.size() != expectedRemainingTopicsAfterCleanup.size());
-        } finally {
-            if (zkUtils != null) {
-                zkUtils.close();
-            }
+            ListTopicsOptions listTopicsOptions = new ListTopicsOptions();
+            listTopicsOptions.listInternal(true);
+            allTopics.addAll(kafkaAdminClient.listTopics(listTopicsOptions).names().get(30000, TimeUnit.MILLISECONDS));
+            assertThat(allTopics, equalTo(expectedRemainingTopicsAfterCleanup));
```

Comment:
Is this safe here? The future that is returned from `KafkaAdminClient` ensure that the topic was delete? Ie. "delete topic" is not async anymore?

Reply:
Yes, I believe so.  In `StreamsResetter` we check each returned `KafkaFuture` for successful completion (lines 347- 354) if it failed for whatever reason the resetter would return with an error return code and fail the test before confirming deletions.  Does this answer your question?

Reply:
Yes. Thanks for confirming.

I was not sure was `entry.getValue().get(30, TimeUnit.SECONDS)` return -- if I understand you correctly, it either returns "success" or it throws. There is no "failed" return code in the future.